prometheus/Readme.md

```diff
@@ -0,0 +1,62 @@
+# Prometheus Configuration
+
+## Default `query-max-concurrency` Setting: **4**
+
+The `query-max-concurrency` setting in Prometheus controls the maximum number of queries that can be executed concurrently.
+Setting this value appropriately is crucial to ensure that Prometheus can handle the load without overwhelming the system’s resources (CPU, memory, and disk).
+
+### Why Set `query-max-concurrency` to **4**?
+
+A concurrency setting of **4** is a reasonable starting point for most environments with moderate resource availability. Here's why:
+Limiting concurrency helps avoid excessive CPU or memory usage, especially during periods of high query traffic. Prometheus will still be able to handle other tasks (like data ingestion and internal processing) without degrading system performance.
+
+### What Happens if the Concurrency Limit is Exceeded?
+
+If the number of concurrent queries exceeds the `query-max-concurrency` setting , Prometheus will queue excess queries until a slot becomes available. Here's what happens:
+
+1. **Queuing of Queries:**
+   - Queries beyond the concurrency limit will be placed in a queue. They will wait until one of the currently running queries completes and frees up a slot.
+
+2. **Increased Latency:**
+   - When queries are queued, response times will increase, as clients will need to wait for an available slot. This can result in noticeable delays, especially if the queue becomes large.
+
+3. **Potential Timeouts:**
+   - If the query stays in the queue for too long and isn't executed within a reasonable timeframe, it may time out. This can lead to errors or failures in the client making the request
+
+4. **System Load:**
```

Comment:
I think it could strain memory, eventually, but I don't think we will actually see that as an issue. Is the queue of unlimited length?

I don't see how it can strain other resources.